src/client/pkg/require/require.go

```diff
@@ -95,48 +95,234 @@ func NotEqual(tb testing.TB, expected interface{}, actual interface{}, msgAndArg
 	}
 }
 
-// EqualOneOf checks if a value is equal to one of the elements of a slice.
-func EqualOneOf(tb testing.TB, expecteds []interface{}, actual interface{}, msgAndArgs ...interface{}) {
-	tb.Helper()
-	equal := false
-	for _, expected := range expecteds {
-		if reflect.DeepEqual(expected, actual) {
-			equal = true
-			break
+// elementsEqualOrErr is a helper for ElementsEqualOrError and ElementsEqual
+// that implements their essential logic (comparing the values of 'expecteds'
+// and 'actuals).  The 't' argument is optional (may be set to nil), so that
+// ElementsEqual can pass it, and its error's line number will be set
+// appropriately, while ElementsEqualOrError (which does not accept a 't'
+// argument' can pass nil)
+//
+// As described in the comments for 'ElementsEqual()', if the elements of
+// 'expecteds' and 'actuals' are pointers, elementsEqualOrErr will unwrap the
+// pointers before comparing them, and 'nil' and the empty slice are treated as
+// equivalent
+func elementsEqualOrErr(tb testing.TB, expecteds interface{}, actuals interface{}) error {
+	if tb != nil {
+		tb.Helper()
```

Comment:
Does this actually do anything? I though that `tb.Helper` would only be relevant if the function (or a function it calls) might call `t.Fatal` in which case it gets omitted from backtraces. But it doesn't look like this function can call `t.Fatal`

Reply:
Nope, I meant to remove it in the other PR too, so I'll definitely kill it here